pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -109,12 +109,16 @@ func (h *HeapsterMetricsClient) GetCPUUtilization(namespace string, selector map
 	return &utilization, timestamp, nil
 }
 
-func (h *HeapsterMetricsClient) GetCpuConsumptionAndRequestInMillis(namespace string, selector map[string]string) (avgConsumption int64,
+func (h *HeapsterMetricsClient) GetCpuConsumptionAndRequestInMillis(namespace string, selector string) (avgConsumption int64,
 	avgRequest int64, timestamp time.Time, err error) {
 
-	labelSelector := labels.SelectorFromSet(labels.Set(selector))
+	// TODO: Eliminate deserialization/reserialization of the selector.
```

Comment:
@bgrant0607 This defeats the entire purpose of the new selector type-- you have to pass this through unparsed, or make a selector type that does that for you. Otherwise we're still limited by what clients can understand.